src/main/java/com/facebook/openwifirrm/optimizers/tpc/MeasurementBasedApApTPC.java

```diff
@@ -64,6 +64,48 @@ public class MeasurementBasedApApTPC extends TPC {
 	/** Nth smallest RSSI (zero-indexed) is used for Tx power calculation */
 	private final int nthSmallestRssi; // TODO non-zero values untested
 
+	/** Factory method to parse generic args map into the proper constructor */
+	public static MeasurementBasedApApTPC makeWithArgs(
+		DataModel model,
+		String zone,
+		DeviceDataManager deviceDataManager,
+		Map<String, String> args
+	) {
+		int coverageThreshold = DEFAULT_COVERAGE_THRESHOLD;
+		int nthSmallestRssi = DEFAULT_NTH_SMALLEST_RSSI;
+
+		String arg;
+		if ((arg = args.get("coverageThreshold")) != null) {
+			try {
+				coverageThreshold = Integer.parseInt(arg);
+			} catch (NumberFormatException e) {
+				logger.error(
+					"Invalid integer passed to parameter coverageThreshold, using default value: ",
```

Comment:
nit: for `logger.error` you don't need to end the string with `: `, it prints the exception in a readable way already